### DISCUSSION TITLE
Problem with relation when deleting

### DISCUSSION THREAD

**Discussion by maverpolo022** [prisma/prisma#16142] [Q&A] [answered]
So, I have one model that goes like this
```
model Author {
    id String @id @default(cuid())
    Posts Post[]
}
```
```
model Post {
    ...
   author Author @relation(fields: [authorId], references: [id])
   authorid String
}
```

Here's the scenario (may not make sense because this is hypothetical but exactly what I need),
1. When an author is deleted, I want to still keep their posts. It would be better if I can also keep their id because i will check if the author is present before further use.

[this is the part causing error in my app]
[for this, i don't know if relational appraoch is better or i just simply hardcode values in both the tables, but then i may have to deal with `.update` time complexity]

2. fyi, a post once created will never be deleted in my use-case

**Comment by maverpolo022** [comment, marked as answer]
Actually solved this issue by editing to
```
model Post {
    ...
   author Author? @relation(fields: [authorId], references: [id])
   authorid String?
}
```

this way, if an author is deleted instead of giving out an error, it simply blanks out the authorid field in Post table.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
